src/components/BounceRules/index.jsx

```diff
@@ -0,0 +1,233 @@
+import React from "react";
+import { Redirect } from "react-router-dom";
+import "./BounceRules.css";
+import axios from "axios";
+
+export default class BounceRules extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      searchToken: "",
+      isRedirectingToDetail: false,
+      selectedRule: {},
+      rules: [],
+      pageIndex: 1,
+      pageInterval: 10,
+    };
+
+    this.updateSearchToken = this.updateSearchToken.bind(this);
+    this.handleKeyDown = this.handleKeyDown.bind(this);
+  }
+
+  async componentDidMount() {
+    const { data, status } = await axios.get(
```

Comment:
Currently as constructed with network requests being called directly in the topmost stateful component, it's not unit testable for all the parts of the app. We can construct it in a way to have the topmost stateful component manage all the state, abstract out the api calls to another file so we can mock them with jest or you can use the mock axios library to integration test these top level components and the next components down are for the most part stateless functional components that we can unit test based on different props given and add storybook stories for to show all the design states based on different props